src/components/app.jsx

```diff
@@ -1,18 +1,35 @@
 import React from 'react';
 import Header from './header/header';
-import contacts from '../data';
+import contactsData from '../data';
 import ToolBar from './toolBar/toolBar';
 import ContactList from './contactList/contactList';
 
 
 export default class App extends React.PureComponent {
+  state = {
+    contacts: contactsData,
+  };
+
+  handleInputChange = (evt) => {
+    const value = evt.target.value;
+
+    const filteredContacts = contactsData.filter(contact => {
+      const name = contact.name.toLowerCase().replace(/ /g, '');
+      return name.indexOf(value.toLowerCase()) !== -1;
+    });
+
+    this.setState({
+      contacts: filteredContacts,
+    });
+  }
+
   render() {
     return (
       <div className="container">
         <Header />
         <main>
-          <ToolBar />
-          <ContactList contacts={contacts} />
+          <ToolBar inputChangeHandler={this.handleInputChange} />
```

Comment:
Usually naming is `onInputChange={this.handleInputChange}`

Reply:
Done